x/gamm/keeper/pool_service_test.go

```diff
@@ -222,6 +242,181 @@ func (suite *KeeperTestSuite) TestCreateBalancerPool() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestInitializePool() {
+	params := suite.App.GAMMKeeper.GetParams(suite.Ctx)
+	testAccount := suite.TestAccs[0]
+
+	// get raw pool creation fee(s) as DecCoins
+	poolCreationFeeDecCoins := sdk.DecCoins{}
+	for _, coin := range params.PoolCreationFee {
+		poolCreationFeeDecCoins = poolCreationFeeDecCoins.Add(sdk.NewDecCoin(coin.Denom, coin.Amount))
+	}
+
+	tests := []struct {
+		name        string
+		msg         swaproutertypes.CreatePoolMsg
+		usingMock   bool
+		emptySender bool
+		expectPass  bool
+	}{
+		{
+			name:        "initialize balancer pool with default assets",
+			msg:         balancer.NewMsgCreateBalancerPool(testAccount, defaultPoolParams, defaultPoolAssets, defaultFutureGovernor),
+			emptySender: false,
+			expectPass:  true,
+		},
+		{
+			name:        "initialize stableswap pool with default assets",
+			msg:         stableswap.NewMsgCreateStableswapPool(testAccount, defaultStableSwapPoolParams, defaultStableSwapPoolAssets, defaultScalingFactor, ""),
+			emptySender: false,
+			expectPass:  true,
+		},
+		{
+			name:      "initialize a CL pool which cause panic",
+			msg:       balancer.NewMsgCreateBalancerPool(testAccount, defaultPoolParams, defaultPoolAssets, defaultFutureGovernor),
+			usingMock: true,
+		},
+	}
+
+	for _, test := range tests {
+		suite.SetupTest()
```

Comment:
Can you please use `suite.Run` similar to: https://github.com/osmosis-labs/osmosis/blob/f611e01c7fb9ab9d5348250cb0b65e08f3ec651b/x/concentrated-liquidity/pool_test.go#L74